### PR TITLE
fix `extract_signature` to work with sql.Composable objects

### PR DESCRIPTION
**Pull request by beniwohli**
see http://initd.org/psycopg/docs/sql.html

### CODE REVIEW

**Comment by beniwohli** [comment]
jenkins test this

**Comment by beniwohli** [comment]
jenkins test this

File: elasticapm/instrumentation/packages/dbapi2.py [hunk 1]
```diff
@@ -170,10 +175,17 @@ def executemany(self, sql, param_list):
         return self._trace_sql(self.__wrapped__.executemany, sql,
                                param_list)
 
+    def _bake_sql(self, sql):
+        """
+        Method to turn the "sql" argument into a string
+        """
+        return sql
```
Comment (simitt) [inline elasticapm/instrumentation/packages/dbapi2.py]: The method description might be a bit confusing, as it actually does nothing in this class.
Would it do any harm if you'd actually move the `as_string` check here?

File: elasticapm/instrumentation/packages/dbapi2.py [hunk 2]
```diff
@@ -121,6 +121,11 @@ def scan(tokens):
 
 
 def extract_signature(sql):
```
Comment (simitt) [inline elasticapm/instrumentation/packages/dbapi2.py]: Not part of this PR, but out of interest, why is this not a method of `CursorProxy` ? Afaics this is only used within sql cursor proxy implementations.

File: elasticapm/instrumentation/packages/dbapi2.py (hunk 1 above)
Comment (beniwohli) [reply elasticapm/instrumentation/packages/dbapi2.py]: the `as_string` check is specific to psycopg2 (the Python PostgreSQL driver), so I think putting it in the psycopg2-specific subclass makes sense. However I agree that the docstring needs work.

File: elasticapm/instrumentation/packages/dbapi2.py (hunk 2 above)
Comment (beniwohli) [reply elasticapm/instrumentation/packages/dbapi2.py]: I thought about this as well, but into the other direction: remove `CursorProxy.extract_signature` and call `dbapi2.extract_signature()` directly where we need it.

I guess the idea was that if we instrument a SQL backend that needs a different parser, we could plug it in there. @roncohen might remember.

Comment (simitt) [reply elasticapm/instrumentation/packages/dbapi2.py]: Thank you for giving some context. Might be worth giving it a thought at some point, but as mentioned unrelated to this PR.
